test/e2e/storage/testsuites/volume_modify.go

```diff
@@ -0,0 +1,310 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuites
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	storagev1alpha1 "k8s.io/api/storage/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/discovery"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/feature"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	e2evolume "k8s.io/kubernetes/test/e2e/framework/volume"
+	storageframework "k8s.io/kubernetes/test/e2e/storage/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+const (
+	modifyPollInterval               = 2 * time.Second
+	setVACWaitPeriod                 = 30 * time.Second
+	modifyingConditionSyncWaitPeriod = 2 * time.Minute
+	modifyVolumeWaitPeriod           = 10 * time.Minute
+	vacCleanupWaitPeriod             = 30 * time.Second
+)
+
+type volumeModifyTestSuite struct {
+	tsInfo storageframework.TestSuiteInfo
+}
+
+// InitCustomVolumeModifyTestSuite returns volumeModifyTestSuite that implements TestSuite interface
+// using custom test patterns
+func InitCustomVolumeModifyTestSuite(patterns []storageframework.TestPattern) storageframework.TestSuite {
+	return &volumeModifyTestSuite{
+		tsInfo: storageframework.TestSuiteInfo{
+			Name:         "volume-modify",
+			TestPatterns: patterns,
+			SupportedSizeRange: e2evolume.SizeRange{
+				Min: "1Gi",
+			},
+			TestTags: []interface{}{feature.VolumeAttributesClass},
```

Comment:
Please use `framework.WithFearure` and  `framework.WithFearureGate` in TestTags.

Reply:
`feature.VolumeAttributesClass` already is a `framework.WithFeature`, added a `framework.WithFearureGate` as well